### PR TITLE
INC-838: total of Overdue Prisoner Review

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/INC-838

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt [hunk 1]
```diff
@@ -50,6 +52,7 @@ class IncentiveReviewsService(
     if (bookingIdsMissingIepReviews.isNotEmpty()) {
       throw ListOfDataNotFoundException(bookingIdsMissingIepReviews)
     }
+    val overdueCount = nextReviewDates.map { r -> r.value.nextReviewDate }.count { it.isBefore(LocalDate.now(clock)) }
```
Comment (ushkarev) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt:55]: this only counts the number overdue on this page

Comment (neilmendum) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt:55]: D'oh! I see what you mean as we're querying `offenderSearchService.findOffenders` for a `page`

We have [INC-850](https://dsdmoj.atlassian.net/browse/INC-850) for storing it in the DB which as you commented we may not have sufficient information for. I think we'll know everything we need once we finish [INC-816](https://dsdmoj.atlassian.net/browse/INC-816), so we can park this until then?

Comment (ushkarev) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt:55]: yeah i'm totes ok with a lot of the planned functionality to be part-way-there cos so much of it depends on other parts being done

**Review by ushkarev — APPROVED** [review]
`overdueCount` uses too narrow a query to count overdue reviews, but that's probs ok for now. there are many other changes still needed to feed into this number to make it right anyway

**Pull request merged**